plugins/jvm-abi-gen/src/org/jetbrains/kotlin/jvm/abi/JvmAbiClassBuilderInterceptor.kt

```diff
@@ -86,11 +98,10 @@ class JvmAbiClassBuilderInterceptor(
         ) {
             // Always keep annotation classes
             // TODO: Investigate whether there are cases where we can remove annotation classes from the ABI.
-            if (access and Opcodes.ACC_ANNOTATION != 0) {
-                publicAbi = true
-            }
+            keepClassAsIs = keepClassAsIs || access and Opcodes.ACC_ANNOTATION != 0
 
             internalName = name
+            superInterfaces = interfaces.asList()
```

Comment:
Initially I tried to avoid a unnecessary copy here since we're going to copy the array to the builder at the end of the class processing anyways. Did you do it only as a defensive approach here?

Reply:
I did it only because using lists is preferable to using arrays in Kotlin. It's not a copy, `asList` creates a view, which is supposed to be cheap.

Reply:
Read it as `toList()` my bad. That makes total sense!